### PR TITLE
Adapts the open-telemetry example for using agents

### PR DESCRIPTION
**Pull request by fedefernandez**
Based on the Demo example from [open-telemetry/opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib), this PR updates the example to show how to deploy the collector in Agent mode. 

> The agent collector deployment pattern consists of applications — [instrumented](https://opentelemetry.io/docs/instrumentation/) with an OpenTelemetry SDK using [OpenTelemetry protocol (OTLP)](https://opentelemetry.io/docs/specs/otel/protocol/) — or other [collector](https://opentelemetry.io/docs/collector/)s (using the OTLP exporter) that send telemetry signals to a collector instance running with the application or on the same host as the application (such as a sidecar or a daemonset).
> 
> Each client-side SDK or downstream collector is configured with a collector location:
> 
>
> otel-agent-sdk
>
> Decentralized collector deployment concept
> In the app, the SDK is configured to send OTLP data to a collector.
> The collector is configured to send telemetry data to one or more backends.

[Ref docs](https://opentelemetry.io/docs/collector/deployment/agent/)

The `Animal.kt` client uses the opentelemetry library for instrumentation and for sending telemetry data to the opentelemetry collector.

This example presents the typical flow of observability data where multiple OpenTelemetry Collectors can be deployed.

```mermaid
graph TD;
    ClientA-->Collector;
    ClientB-->Collector;
    Collector-->Jaeger;
    Collector-->Zipkin;
    Collector-->Prometheus;
```

- Clients send data directly to the OpenTelemetry Collector
- The OpenTelemetry Collector Collector then sends the data to the appropriate backend. In the example Jaeger, Zipkin, and Prometheus.

The example uses `docker-compose`. To run the demo, switch to the `server/docker/opentelemetry` folder and run:

```shell
docker compose up -d
```

Docker should expose the following backends:

- Jaeger at http://0.0.0.0:16686
- Zipkin at http://0.0.0.0:9411
- Prometheus at http://0.0.0.0:9090

### CODE REVIEW

**Review by Montagon — APPROVED** [review]
Great job! 👏

File: server/docker/opentelemetry/otel-collector-config.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  extensions: [pprof, zpages, health_check]
+  pipelines:
+    traces:
+      receivers: [otlp]
+      processors: [batch]
+      exporters: [zipkin, otlp]
+    metrics:
+      receivers: [otlp]
+      processors: [batch]
+      exporters: [prometheus]
```
Comment (raulraja) [inline server/docker/opentelemetry/otel-collector-config.yaml:41]: We will need this to submit the metric to Datadog for our current client needs.

**Review by raulraja — APPROVED** [review]
Thank you @fedefernandez !

File: server/docker/opentelemetry/otel-collector-config.yaml (hunk 1 above)
Comment (fedefernandez) [reply server/docker/opentelemetry/otel-collector-config.yaml:41]: I will do some tests for the DataDog integration after this. Traces and logs can be sent, too.

**Pull request merged**